src/regExpressions.test.ts

```diff
@@ -64,3 +70,41 @@ describe('captureFullUrlParts', () => {
         expect(dateTimeWithTimeZoneRegExp.test(timeWithoutTimeZone)).toBeFalsy();
     });
 });
+
+describe('captureResourceTypeRegExp', () => {
+    test('Success match', () => {
+        const resourceString = `Patient/12345678`;
+        const actualMatch = resourceString.match(captureResourceTypeRegExp);
+
+        const expectedMatch = ['Patient/12345678', 'Patient'];
+
+        for (let i = 0; i < expectedMatch.length; i += 1) {
+            // @ts-ignore
+            expect(actualMatch[i]).toEqual(expectedMatch[i]);
+        }
```

Comment:
Couldn't you just do `expect(actualMatch).toEqual(expectedMatch)`

Reply:
The actualMatch is actually longer than the expectedMatch, with additional metadata fields like 'input', and those are the fields that we don't care.

Reply:
Ah, can we use slice then?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

`expect(actualMatch).toEqual(expectedMatch.slice(expectedMatch.length))`

Reply:
Emmm that could work. I copied the for loop check from other test and didn't update it.

Reply:
Emmm it didn't work actually. The resulted actualMatch result is of type RegExpMatchArray, which is extended from Array. But surprisingly slice does not work on it. I'll keep it as is.